ecies.go

```diff
@@ -263,16 +263,20 @@ func (prv *PrivateKey) Decrypt(rand io.Reader, c, s1, s2 []byte) (m []byte, err
 	)
 
 	switch c[0] {
-	case 2, 3, 4:
-		rLen = ((prv.PublicKey.Curve.Params().BitSize + 7) / 4)
-		if len(c) < (rLen + hLen + 1) {
-			err = ErrInvalidMessage
-			return
-		}
+	case 2, 3:
+		// https://github.com/golang/go/blob/go1.19.5/src/crypto/elliptic/elliptic.go#L147
+		rLen = 1 + (prv.PublicKey.Curve.Params().BitSize+7)/8
+	case 4:
+		// https://github.com/golang/go/blob/go1.19.5/src/crypto/elliptic/elliptic.go#L120
+		rLen = 1 + 2*((prv.PublicKey.Curve.Params().BitSize+7)/8)
 	default:
 		err = ErrInvalidPublicKey
 		return
 	}
+	if len(c) < (rLen + hLen + 1) {
```

Comment:
I would code it a bit different: use `kLen  = prv.PublicKey.Curve.Params().BitSize+7)/8` then instead of rLen I would use `mStart = 1 + kLen` in the non-compressed format and `mStart = 1 + 2 * kLen` in the compressed format...

Reply:
other than that looks good to me!

Reply:
I wanted the change to be minimalistic... but reusing `kLen  = prv.PublicKey.Curve.Params().BitSize+7)/8` makes sense to me - I will do that.

The `rLen` is used down in the below code, so semantically it makes sense to have both `mStart` and `rLen`.

Also at line 276 I guess `+ 1` is to accommodate for at least 1 character of the encrypted message itself. I'm not sure if the empty message is a good edge case to support. So will leave that as is in the original too.

Reply:
@ldts updated in force-push.

I actually found a way to part with rLen easily while keeping the code semantics intact.